google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java

```diff
@@ -865,9 +862,12 @@ public HttpResponse execute() throws IOException {
     Preconditions.checkNotNull(requestMethod);
     Preconditions.checkNotNull(url);
 
-    Span span = tracer.spanBuilder(traceSpanNamePrefix + "execute").startSpan();
+    Span span = tracer
+        .spanBuilder(OpenCensusUtils.SPAN_NAME_HTTP_REQUEST_EXECUTE)
+        .setRecordEvents(OpenCensusUtils.isRecordEvent())
+        .startSpan();
     do {
-      span.addAnnotation("retry #" + numRetries);
+      span.addAnnotation("retry #" + (numRetries - retriesRemaining));
```

Comment:
probably for the number of retries you can use an annotation with an attribute, it has the type int.